src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -89,53 +89,144 @@
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
+ *
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // OK, no overlap
+ *     foo(arg);         // OK, no overlap
+ *   }
+ *
+ *   public void foo(int c) {
+ *     // overlapping by method argument
+ *     c = c;            // violation, reference to instance variable "c" requires "this"
+ *   }
+ * }
+ * </pre>
  * <p>
  * To configure to check the {@code this} qualifier for fields only:
  * </p>
+ * <p>(with validateOnlyOverlapping set to false)</p>
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;&gt;
  *   &lt;property name=&quot;checkMethods&quot; value=&quot;false&quot;/&gt;
+ *   &lt;property name=&quot;validateOnlyOverlapping&quot; value=&quot;false&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
+ *
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // violation, reference to instance variable "b" requires "this"
+ *     foo(arg);         // OK, no validation for methods
+ *   }
+ *
+ *   public void foo(int c) {
+ *     // overlapping by method argument
+ *     c = c;            // violation, reference to instance variable "c" requires "this"
+ *   }
+ * }
+ * </pre>
  * <p>
- * Examples of how the check works if validateOnlyOverlapping option is set to true:
+ * To configure to check the {@code this} qualifier for fields only:
  * </p>
+ * <p>(with validateOnlyOverlapping set to true)</p>
```

Comment:
what is purpose of this paragraph? It is clear that this property is set to false when you see config.
But it is not clear how this changes anything

Reply:
I have not included any example for **just** validateOnlyOverlapping, as it was easier to demonstrate the usage of property with other properties. That's why I thought I should explicitly mention the value, so that user does not miss it.

Reply:
May be it is better to move it to previous paragraph (to configure check ...)

Also, there is a typo now
`To configure to check`  -> To configure check

Reply:
Yes, I thought of moving it into the previous paragraph, but in xml doc, I don't know how a reliable way to insert newline, so that it appears like this -
```
To configure check the this qualifier for fields only:
(with validateOnlyOverlapping set to true)
```

instead of looking like this -
```
To configure check the this qualifier for fields only (with validateOnlyOverlapping set to true):
```

Reply:
it can be 
`To configure check for fields only (with validateOnlyOverlapping = true):

Reply:
Done